app/controllers/businessactivities/WhatYouNeedController.scala

```diff
@@ -34,6 +42,13 @@ import scala.concurrent.Future
 
   def get = authAction.async {
     implicit request =>
-      Future.successful(Ok(what_you_need(routes.InvolvedInOtherController.get().url)))
+      dataCacheConnector.fetch[BusinessMatching](request.credId, BusinessMatching.key) map { businessMatching =>
+        (for {
+          bm <- businessMatching
+          ba <- bm.activities
+        } yield {
+          Ok(what_you_need(routes.InvolvedInOtherController.get().url, Some(ba)))
+        }) getOrElse Redirect(controllers.routes.RegistrationProgressController.get())
```

Comment:
Just interested. Why Registration Progress Controller if we fail to get the activities? 

Do we want to do this? Is this error recoverable? If not we likely want to raise an exception such that we can investigate. If we encounter a non recoverable error such as failing to locate business matching in the cache then sending the user back to the progress page might be confusing? The user will keep trying to get into the section and could get stuck in a loop without an exception being thrown therefore giving us nothing to investigate. 

If we need this data to be able to proceed then should we raise an error if this data can't be located? 

The alternative is to load a default if the data can't be retrieved.

Reply:
Done